external/CapnProto.cmake

```diff
@@ -19,71 +19,87 @@
 # http://numenta.org/licenses/
 # -----------------------------------------------------------------------------
 
-option(FIND_CAPNP "Use preinstalled Cap'n Proto." OFF)
-
-if (${FIND_CAPNP})
-  find_package(CapnProto)
-  # Most CAPNP* variables are set correctly but make sure we have the
-  # static libraries.
-  find_library(LIB_KJ ${STATIC_PRE}kj${STATIC_SUF})
-  find_library(LIB_CAPNP ${STATIC_PRE}capnp${STATIC_SUF})
-  find_library(LIB_CAPNPC ${STATIC_PRE}capnpc${STATIC_SUF})
-  set(CAPNP_LIBRARIES ${LIB_CAPNPC} ${LIB_CAPNP} ${LIB_KJ})
-  set(CAPNP_LIBRARIES_LITE ${LIB_CAPNP} ${LIB_KJ})
-
-  # Create a dummy target to depend on.
-  add_custom_target(CapnProto)
-
-else ()
-  # Build Cap'n Proto from source.
-  if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
-    set(CAPNP_DEFINITIONS "-DCAPNP_LITE=1 -DEXTERNAL_CAPNP=1")
-    set(CAPNP_CXX_FLAGS "-m${BITNESS}")
-  else()
-    set(CAPNP_DEFINITIONS "-DCAPNP_LITE=0")
-    set(CAPNP_CXX_FLAGS "-fPIC -m${BITNESS}")
-  endif()
-  ExternalProject_Add(
-    CapnProto
-    GIT_REPOSITORY https://github.com/sandstorm-io/capnproto.git
-    GIT_TAG v0.5.3
-    UPDATE_COMMAND ""
-    CONFIGURE_COMMAND
-        ${CMAKE_COMMAND}
-        ${CAPNP_DEFINITIONS}
-        -DCMAKE_CXX_FLAGS=${CAPNP_CXX_FLAGS}
-        -DBUILD_TESTING=OFF
-        -DBUILD_SHARED_LIBS=OFF
-        -DCMAKE_INSTALL_PREFIX=${EP_BASE}/Install
-        -G "${CMAKE_GENERATOR}"
-        ${EP_BASE}/Source/CapnProto/c++
-  )
+# Build Cap'n Proto from source.
+#
+# OUTPUTS
+#
+# CAPNP_LINK_LIBRARIES: list of capnproto libraries (paths) needed by apps for linking
+# CAPNP_INCLUDE_DIRS
+# CAPNP_EXECUTABLE
+# CAPNPC_CXX_EXECUTABLE
+# CAPNP_CMAKE_DEFINITIONS: informational; platform-specific cmake defintions used by capnproto build
+# CAPNP_COMPILER_DEFINITIONS: list of -D compiler defintions needed by apps that
+#                             are built against this library (e.g., -DCAPNP_LITE)
+
+
+set(capnproto_lib_url "${REPOSITORY_DIR}/external/common/share/capnproto/capnproto-c++-0.5.3.tar.gz")
+set(capnproto_win32_tools_url "${REPOSITORY_DIR}/external/common/share/capnproto/capnproto-c++-win32-0.5.3.zip")
```

Comment:
@breznak, are you recommending defining a variable `set(version "0.5.3") and then using it like this:

```
set(capnproto_lib_url "${REPOSITORY_DIR}/external/common/share/capnproto/capnproto-c++-${version}.tar.gz")
```

If so, it's not worth it, since the rest of the naming scheme is not consistent in the capnp's other tagged releases in their github.